src/pynwb/file.py

```diff
@@ -552,7 +559,7 @@ def objects(self):
 
     @property
     def modules(self):
-        warn("NWBFile.modules has been replaced by NWBFile.processing.", DeprecationWarning)
+        self._error_on_new_warn_on_construct(error_msg="NWBFile.modules is deprecated. Use NWBFile.processing instead.")
```

Comment:
These and the below properties are never in construct mode. Since they have already been deprecated, I think we can just remove the functions.

Reply:
Removed these properties. `icephys_filtering` should remain since it can be used in construct mode, correct?